### PR TITLE
Update repository object in UpdateObjectService

### LINKED ISSUE #4838: Update RepositoryObject is UpdateObjectService
When a RepositoryObject exists for the object, update it. This is in addition to updating the Dro/Collection/AdminPolicy via CocinaObjectStore.store.

There is NO feature flag for this operation.

### PR DESCRIPTION
**Pull request by jcoyne**
## Why was this change made? 🤔

Fixes #4838

## How was this change tested? 🤨

⚡ ⚠ If this change has cross service impact, including data writes to shared file systems, ***run [integration tests](https://github.com/sul-dlss/infrastructure-integration-test)*** and/or test in [stage|qa] environment, in addition to specs. ⚡

### CODE REVIEW

File: app/services/update_object_service.rb [hunk 1]
```diff
@@ -35,6 +35,9 @@ def update
 
     cocina_object_without_metadata = Cocina::Models.without_metadata(cocina_object)
 
+    # TODO: after migration to RepositoryObject we can remove these nil checks
+    RepositoryObject.find_by(external_identifier: cocina_object.externalIdentifier)&.head_version&.update!(RepositoryObjectVersion.to_model_hash(cocina_object_without_metadata))
+
```
Comment (mjgiarlo) [inline app/services/update_object_service.rb:40]: Do we always want to patch the updates into the head version? I.e., what if the head version is already closed? @jcoyne @justinlittman

Comment (mjgiarlo) [reply app/services/update_object_service.rb:40]: Sounds good to me.

Comment (justinlittman) [reply app/services/update_object_service.rb:40]: Should not automatically open versions.

Comment (justinlittman) [reply app/services/update_object_service.rb:40]: Also, update service will be verifying that object is open.

**Review by justinlittman — APPROVED** [review]

**Pull request merged**